src/fix_llvm_assert.h

```diff
@@ -0,0 +1,16 @@
+// This file is a part of Julia. License is MIT: http://julialang.org/license
+
+// Include this file after every blocks of LLVM includes to set the assertion back.
+
+#ifdef NDEBUG
+#  ifndef JL_NDEBUG
+#    undef NDEBUG
+#    include <assert.h>
+// Set NDEBUG back so that we can include another LLVM header right after
+#    define NDEBUG
+#  endif
+#else
+#  ifdef JL_NDEBUG
+#    undef JL_NDEBUG
```

Comment:
No, the organization is such that our own assertion is enabled when either LLVM or us enables it. Then `NDEBUG` is always respecting the LLVM state andd `JL_NDEBUG` is always the correct thing to check to enable our own assertion so if LLVM enables the assertion, this logic makes sure the assertion is enabled in the release build.

Only exception being the one in `julia.h` where I also want to repect user setting for embedding usage.